src/daft-proto/proto/v1/daft.proto

```diff
@@ -318,45 +318,62 @@ message IfElse {
   Expr predicate = 3;
 }
 
-message Function {
-  oneof descriptor {
-    RsFunction rs = 1;
-    PyFunction py = 2;
+message ScalarFn {
+  oneof variant {
+    BuiltinFn builtin = 1;
+    PyFn py = 2;
   }
-  Args args = 3;
 
   // For rust functions (builtins), we only have simple names.
-  message RsFunction {
+  message BuiltinFn {
     string name = 1;
-  }
+    Args args = 2;
 
-  // TODO: Daft does not handle static parameter binding.
-  message Args {
-    repeated Arg args = 1;
-  }
+    // TODO: Daft does not handle static parameter binding.
+    message Args {
+      repeated Arg args = 1;
+    }
 
```

Comment:
You need to keep the field numbering the same, and there are extra spaces/formatting issues.